tests/Integration/UI/SettingsPageTest.php

```diff
@@ -193,6 +207,136 @@ public function test_invalid_api_secret_values_are_emptied_when_validated(): voi
 		self::assertSame( $expected, $actual );
 	}
 
+	/**
+	 * Verifies that valid API credentials are retained when validated with the Validation API
+	 *
+	 * @since 3.11.0
+	 *
+	 * @covers \Parsely\UI\Settings_Page::validate_basic_section
+	 * @uses   \Parsely\Parsely::__construct
+	 * @uses   \Parsely\Parsely::api_secret_is_set
+	 * @uses   \Parsely\Parsely::are_credentials_managed
+	 * @uses   \Parsely\Parsely::get_api_secret
+	 * @uses   \Parsely\Parsely::get_options
+	 * @uses   \Parsely\Parsely::get_managed_credentials
+	 * @uses   \Parsely\UI\Settings_Page::__construct
+	 * @uses   \Parsely\UI\Settings_Page::get_logo_default
+	 * @uses   \Parsely\UI\Settings_Page::get_obfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::get_unobfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::sanitize_site_id
+	 * @uses   \Parsely\UI\Settings_Page::validate_advanced_section
+	 * @uses   \Parsely\UI\Settings_Page::validate_options
+	 * @uses   \Parsely\UI\Settings_Page::validate_options_post_type_tracking
+	 * @uses   \Parsely\UI\Settings_Page::validate_recrawl_section
+	 * @uses   \Parsely\Validator::validate_api_secret
+	 * @uses   \Parsely\Validator::validate_site_id
+	 * @uses   \Parsely\Validator::validate_api_credentials
+	 *
+	 * @group settings-page
+	 * @group settings-page-validation
+	 */
+	public function test_valid_api_credentials_are_retained_when_validated(): void {
+		$options = self::$parsely->get_options();
+
+		$options['apikey']     = 'mydomain.com';
+		$options['api_secret'] = 'valid_api_secret_key_based_on_length';
```

Comment:
Question: With the new changes, does it matter what value we put in `$options['api_secret']`? It would succeed regardless of the value due to `mock_request_api_credentials_validation_success`, right?

Reply:
If we keep the existing `validate_*` functions, then yes, since if you enter an invalid API secret (less than 30 chars), it will not run the API request to validate the credentials and instead will fail the validation immediately. The same is true for the API Key (If it's not a valid domain - www prefix and up to three dots)